src/test/java/teammates/test/cases/webapi/GetHasResponsesActionTest.java

```diff
@@ -0,0 +1,238 @@
+        assertEquals(messageOutput.getMessage(), "No feedback question with id: fake-question-id");
+    }
+
+    @Test
+    protected void testExecute_getRespondentsInCourse_shouldPass() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor1OfCourse1.googleId);
+
+        testGettingRespondentsInCourse();
+
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        loginAsStudent(student1InCourse1.googleId);
+
+        testGettingRespondentsInCourse();
+    }
+
+    protected void testGettingRespondentsInCourse() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        ______TS("At least 1 respondent");
+        assertTrue(logic.hasResponsesForCourse(instructor1OfCourse1.getCourseId()));
+
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        GetHasResponsesAction getHasResponsesAction = getAction(params);
+        JsonResult jsonResult = getJsonResult(getHasResponsesAction);
+        HasResponsesData hasResponsesData = (HasResponsesData) jsonResult.getOutput();
+
+        assertEquals(jsonResult.getStatusCode(), HttpStatus.SC_OK);
+        assertTrue(hasResponsesData.hasResponses());
+
+        ______TS("Course with 0 respondents");
+
+        InstructorAttributes instructor1OfCourse4 = typicalBundle.instructors.get("instructor1OfCourse4");
+        loginAsInstructor(instructor1OfCourse4.googleId);
+
+        assertFalse(logic.hasResponsesForCourse(instructor1OfCourse4.getCourseId()));
+
+        params = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse4.courseId,
+        };
+
+        getHasResponsesAction = getAction(params);
+        jsonResult = getJsonResult(getHasResponsesAction);
+        hasResponsesData = (HasResponsesData) jsonResult.getOutput();
+
+        assertEquals(jsonResult.getStatusCode(), HttpStatus.SC_OK);
+        assertFalse(hasResponsesData.hasResponses());
+    }
+
+    @Test
+    protected void getRespondentsForQuestion_shouldPass() {
+        ______TS("Question with more than 1 response");
+
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        List<FeedbackSessionAttributes> feedbackSessionAttributesList =
+                logic.getFeedbackSessionsForCourse(instructor1OfCourse1.getCourseId());
+        List<FeedbackQuestionAttributes> feedbackQuestionAttributesList = logic.getFeedbackQuestionsForSession(
+                feedbackSessionAttributesList.get(0).getFeedbackSessionName(), instructor1OfCourse1.getCourseId());
+
+        loginAsInstructor(instructor1OfCourse1.googleId);
+        assertTrue(logic.areThereResponsesForQuestion(feedbackQuestionAttributesList.get(0).getFeedbackQuestionId()));
```

Comment:
You can get the question directly from `dataBundle` and use `logic.getQuestions(session, course, questionNumber)` to retrieve the questionId. The current way is a bit implicit.